cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -837,4 +847,31 @@ private void handleAuthException(Runnable function) {
     public Completable deleteGenericResourceByIdAsync(String databaseServerId) {
         return handleAuthException(() -> azure.genericResources().deleteByIdAsync(databaseServerId));
     }
+
+    private DiskEncryptionSetInner getDiskEncryptionSetInner(String sourceVaultId, String encryptionKeyUrl, String location, Map<String, String> tags) {
+        SourceVault sourceVault = new SourceVault().withId(sourceVaultId);
+        KeyVaultAndKeyReference keyUrl = new KeyVaultAndKeyReference().withKeyUrl(encryptionKeyUrl).withSourceVault(sourceVault);
+        DiskEncryptionSetIdentityType desIdType = new DiskEncryptionSetIdentityType().SYSTEM_ASSIGNED;
+        EncryptionSetIdentity eSetId = new EncryptionSetIdentity().withType(desIdType);
+        DiskEncryptionSetType desType = new DiskEncryptionSetType().ENCRYPTION_AT_REST_WITH_CUSTOMER_KEY;
+        return (DiskEncryptionSetInner) new DiskEncryptionSetInner()
+                .withEncryptionType(desType)
+                .withActiveKey(keyUrl)
+                .withIdentity(eSetId)
+                .withLocation(location)
+                .withTags(tags);
+    }
+
+    public DiskEncryptionSetInner createOrUpdateDiskEncryptionSet(String diskEncryptionSetName, String encryptionKeyUrl, String envName, String location,
+            String resourceGroupName, String sourceVaultId, Map<String, String> tags) {
+        DiskEncryptionSetInner desIn = getDiskEncryptionSetInner(sourceVaultId, encryptionKeyUrl, location, tags);
+        DiskEncryptionSetsInner dSetsIn = azureClientCredentials.getComputeManager().inner().diskEncryptionSets();
+        return dSetsIn.createOrUpdate(resourceGroupName, diskEncryptionSetName, desIn);
+    }
+
+    public String getVaultResourceGroup(String vaultName) {
+        VaultsInner inner = (VaultsInner) ((GroupableResourcesImpl) azureClientCredentials.getKeyVaultManager().vaults()).inner();
+        PagedList<VaultInner> vaultsInSubscription = inner.listBySubscription();
+        return vaultsInSubscription.stream().filter(vInner -> vInner.name().equals(vaultName)).map(VaultInner::id).findFirst().orElse(null);
```

Comment:
Have you verified if this actually works as expected? In particular, `VaultInner.id()` was supposed to contain the whole Resource ID, from which the RG has to be extracted in a separate step. On the other hand, if this is indeed the Resource ID, why not return and use it directly, without extracting the RG part? Then the method name shall be changed to `getVaultResourceIdByVaultName`.

Reply:
@doktoric , @biharitomi, @topolyai5, @bergerdenes Please confirm that we can go with this workaround, given it may have performance hits. The actual impact is unknown at the moment.

Reply:
UPDATE: After some thinking, extra wide permissions (possibly at subscription scope?) might be needed for the above `listBySubscription()` call. It first needs to be confirmed if this is indeed the case. If it turns out to be valid, this approach will have to be dropped for infeasibility since customers will typically not grant subscription-level access.

Reply:
I think the real problem here is the required permission right  @lajosrodek ?

Reply:
@juhi-09 please do the api change for introducing the vault resource group name. We dont want to request full acount permission from customers.